q2as_json.h

```diff
@@ -9,6 +9,139 @@
 
 #include <string_view>
 
+template<typename TargetType, typename SourceType>
+bool q2as_type_in_range(SourceType value)
+{
+    // Prevent bool as TargetType or SourceType to avoid edge cases
+    static_assert(!std::is_same_v<TargetType, bool>, "TargetType cannot be bool");
+    static_assert(!std::is_same_v<SourceType, bool>, "SourceType cannot be bool");
+
+    constexpr bool is_target_integer = std::numeric_limits<TargetType>::is_integer;
+    constexpr bool is_source_integer = std::numeric_limits<SourceType>::is_integer;
+    constexpr bool is_target_signed = std::numeric_limits<TargetType>::is_signed;
+    constexpr bool is_source_signed = std::numeric_limits<SourceType>::is_signed;
+
+    constexpr TargetType max = std::numeric_limits<TargetType>::max();
+    constexpr TargetType min = std::numeric_limits<TargetType>::min();
+
+    // Target and Source are integers
+    if (is_target_integer && is_source_integer)
+    {
+        if (is_target_signed && is_source_signed)
+        {
+            return value <= max && value >= min;
+        }
+
+        if (!is_target_signed && !is_source_signed)
+        {
+            return value <= max;
+        }
+
+        if (is_target_signed && !is_source_signed)
+        {
+            return value <= static_cast<SourceType>(max);
+        }
+
+        if (!is_target_signed && is_source_signed)
+        {
+            return value >= 0 && static_cast<uint64_t>(value) <= static_cast<uint64_t>(max);
+        }
+    }
+
+    // Both floating-point
+    if (!is_target_integer && !is_source_integer)
+    {
+        return value <= max && value >= min;
```

Comment:
for floating types we might want to do something a bit more in-depth. I wonder if just `static_cast<T>(value) != value` would work. We only support float & double anyways so it's just "can this fit in float or not".

Reply:
Fixed.